public/app/features/alerting/unified/components/rules/central-state-history/CentralAlertHistory.tsx

```diff
@@ -0,0 +1,382 @@
+import { css } from '@emotion/css';
+import { hash } from 'immutable';
+import React, { useCallback, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { useMeasure } from 'react-use';
+
+import { GrafanaTheme2, TimeRange } from '@grafana/data';
+import { isFetchError } from '@grafana/runtime';
+import { SceneComponentProps, SceneObjectBase, sceneGraph } from '@grafana/scenes';
+import {
+  Alert,
+  Button,
+  Field,
+  Icon,
+  Input,
+  Label,
+  LoadingBar,
+  Stack,
+  Text,
+  Tooltip,
+  useStyles2,
+  withErrorBoundary,
+} from '@grafana/ui';
+import { EntityNotFound } from 'app/core/components/PageNotFound/EntityNotFound';
+import { Trans, t } from 'app/core/internationalization';
+import {
+  GrafanaAlertStateWithReason,
+  isAlertStateWithReason,
+  isGrafanaAlertState,
+  mapStateWithReasonToBaseState,
+  mapStateWithReasonToReason,
+} from 'app/types/unified-alerting-dto';
+
+import { stateHistoryApi } from '../../../api/stateHistoryApi';
+import { GRAFANA_DATASOURCE_NAME } from '../../../utils/datasource';
+import { stringifyErrorLike } from '../../../utils/misc';
+import { AlertLabels } from '../../AlertLabels';
+import { CollapseToggle } from '../../CollapseToggle';
+import { LogRecord } from '../state-history/common';
+import { useRuleHistoryRecords } from '../state-history/useRuleHistoryRecords';
+
+const LIMIT_EVENTS = 250;
+const STATE_HISTORY_POLLING_INTERVAL = 10 * 1000; // 10 seconds
+
+const HistoryEventsList = ({ timeRange }: { timeRange?: TimeRange }) => {
+  const styles = useStyles2(getStyles);
+
+  // Filter state
+  const [eventsFilter, setEventsFilter] = useState('');
+  // form for filter fields
+  const { register, handleSubmit, reset } = useForm({ defaultValues: { query: '' } }); //  form for search field
+  const from = timeRange?.from.unix();
+  const to = timeRange?.to.unix();
+  const onFilterCleared = useCallback(() => {
+    setEventsFilter('');
+    reset();
+  }, [setEventsFilter, reset]);
+
+  const {
+    currentData: stateHistory,
```

Comment:
I believe we should be using `data` here instead of `currentData`, that way the list view won't flash empty (temporarily) while we fetch new data after the timerange has been updated by the scene.